types/src/crypto/asymmetric_key.rs

```diff
@@ -417,6 +421,19 @@ impl From<SecretKey> for PublicKey {
     }
 }
 
+impl From<&Result<SecretKey, Error>> for PublicKey {
+    fn from(secret_key: &Result<SecretKey, Error>) -> Self {
+        let key = secret_key.as_ref().unwrap();
+        PublicKey::from(key)
+    }
+}
+
+impl From<Result<SecretKey, Error>> for PublicKey {
+    fn from(key: Result<SecretKey, Error>) -> Self {
+        PublicKey::from(key.unwrap())
```

Comment:
These need to be `TryFrom`s, and in general we should never have `From<Result<_, _>>`

Reply:
Agreed, but we should also remove `impl From<&SecretKey> for PublicKey` and `impl From<SecretKey> for PublicKey` since they're effectively constructors which have panicking code.